templates/prometheus.systemd.erb

```diff
@@ -7,8 +7,13 @@ After=basic.target network.target
 [Service]
 User=<%= scope.lookupvar('prometheus::server::user') %>
 Group=<%= scope.lookupvar('prometheus::server::group') %>
+<% if scope.lookupvar('prometheus::server::env_file_path') -%>
+EnvironmentFile=<%= scope.lookupvar('prometheus::server::env_file_path') %>/prometheus
```

Comment:
Does the file need to exist? Otherwise you can use
```suggestion
EnvironmentFile=-<%= scope.lookupvar('prometheus::server::env_file_path') %>/prometheus
```

Also note that systemd deprecated `EnvironmentFile` in favor of drop in files. Then you can use `Environment=ARGS=...` in this file and it can be overridden in a drop in.

Reply:
> Does the file need to exist?

... i think so? but it's better to be safe than sorry i guess... we *do* create the file elsewhere, if my memory is correct, however... so maybe it's better to actually fail if we fail to create it?

> Also note that systemd deprecated EnvironmentFile in favor of drop in files. Then you can use Environment=ARGS=... in this file and it can be overridden in a drop in.

Noted. The point here, however, is to have a single env file that works across init.d and systemd configurations...